solution.cpp

```diff
@@ -22,113 +22,159 @@ using std::string;
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_primes( const int size,
+int write_primes( const int size_of,
                   bool primes[],
                   const string& infilename,
                   const string& outfilename );
 
-/**
- * Read numbers from `infilename`, and if they are composite, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
+
+
+int write_composites( const int size_of,
                       bool primes[],
                       const string& infilename,
                       const string& outfilename );
 
+
+
 // ----------------------------------------------------------------------------
 
 int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
+
+    infilename.open("input.txt");
+
+    int MAX = find_max(infilename);
+    if(MAX == -1)
+    {
+        cout << "Error in find_max()." << endl;
+        return 1;
     }
 
-    bool primes[max+1];
-    sieve(max+1, primes);
+    bool primes[max_num + 1];
+    sieve(max_num + 1, primes);
 
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
+    int ret_value;
+    ret_value = write_primes(total,primes,infilename,outfilename);
+    if(ret_value == -1)
+    {
+        cout << "Error in write_primes()." << endl;
+        return 1;
     }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
+
+    int ret;
+    ret = write_composites(total,primes,infilename,outfilename);
+     if(ret_value == -1)
+    {
+        cout << "Error in write_composites()." << endl;
+        return 1;
     }
 
-    return 0;  // success
+    cout << "Files successfully created and sorted." << endl;
+
+    return 0;
 }
 
 // ----------------------------------------------------------------------------
 
-// TODO: write prototyped functions
+int find_max(ifstream& infilename)
+{
+    infilename.clear();
+    infilename.seekg(0, infilename.beg);
```

Comment:
if `in filename` was created here using a string passed to the function, rather than having the `ifstream` passed (even though it's global?) you wouldn't need to worry about this.